pkg/virtctl/imageupload/imageupload.go

```diff
@@ -206,6 +214,27 @@ func parseArgs(args []string) error {
 	return nil
 }
 
+func normalizeVolumeMode(volumeMode string, blockVolume bool) (string, error) {
```

Comment:
the current code allows a user to write `--block-volume=false --volume-mode=Block` which is contradictory. It should be an error.

I think you should convert `blockVolume` to a string, so `normalizeVolumeMode` can check for this combination. More importantly, I think you should add a unit test form this new function to make sure all combinations are covered.

Reply:
>the current code allows a user to write --block-volume=false --volume-mode=Block which is contradictory. It should be an error.
If take `--block-volume=false` into considerations, the situation become even more complex because `--block-volume` is default to false, without remove the default value of `--block-volume`, specify flag `--volume-mode=Block` always conflict with `--block-volume`.  Remove the default value of `--block-volume` may break users who are using it.

Reply:
Update the code by telling user the value of `block-volume` is overriding by `volume-mode` if both are specified.

Reply:
I'm not a big fan of this fuzzy semantics. If someone is using the new flag, they should not use also the deprecated one.

More importantly, have you seen my suggestion to add a unit test for this function?

Reply:
> More importantly, have you seen my suggestion to add a unit test for this function?

Yes, I see. I'm planning to work on it today.

Reply:
Add a few tests side by the original tests.